sway-lsp/src/capabilities/semantic_tokens.rs

```diff
@@ -71,24 +71,39 @@ fn create_semantic_token(next_token: &Token, prev_token: Option<&Token>) -> Sema
     }
 }
 
-// these values should reflect indexes in `token_types`
-static FUNCTION: u32 = 1;
-static LIBRARY: u32 = 3;
-static VARIABLE: u32 = 9;
-static ENUM: u32 = 10;
-static STRUCT: u32 = 11;
-static TRAIT: u32 = 12;
+/// these values should reflect indexes in `token_types`
+#[repr(u32)]
+enum TokenTypeIndex {
+    _Class,         // 0
+    Function,       // 1
+    _Keyword,       // 2
+    Namespace,      // 3
+    _Operator,      // 4
+    Parameter,      // 5
+    _String,        // 6
+    _Type,          // 7
+    _TypeParameter, // 8
+    Variable,       // 9
+    Enum,           // 10
+    Struct,         // 11
+    Interface,      // 12
+}
```

Comment:
Rather than including variants for the currently unsupported types, I think you could also do something like:

```rust
#[repr(u32)]
enum TokenTypeIndex {
    Function = 1,
    Namespace = 3,
    Parameter = 5,
    Variable = 9,
    Enum = 10,
    Struct = 11,
    Interface = 12,
}
```

Reply:
Agh nice, didn't know that. Updating now.